File: lib/model/network_model.dart
class NetworkModel {
  NetworkModel({
    required this.networkType,
    required this.isConnected,
  });
  final NetworkType networkType;
  final bool isConnected;

  factory NetworkModel.fromMap(Map<String, dynamic> data) {
    return NetworkModel(
      networkType: NetworkType.fromString(data["connectedTo"]),
      isConnected: data["status"] == "Connected",
    );
  }

  @override
  String toString() =>
      'NetworkModel(networkType: ${networkType.rawValue}, isConnected: $isConnected)';
}

enum NetworkType {
  wifi("Wifi"),
  cellular("Cellular"),
  none("None");

  final String rawValue;
  const NetworkType(this.rawValue);

  factory NetworkType.fromString(String val) {
    if (val == wifi.rawValue) return wifi;
    if (val == cellular.rawValue) return cellular;
    return none;
  }
}

File: ios/Classes/NetworkModel.swift
//
//  NetworkModel.swift
//  Runner
//
//  Created by Vinay Sharma on 27/02/24.
//

import Foundation
import Network

class NetworkModel{
    private let connectedTo : NetworkType
    private let status : Bool
    
    init(type : NetworkType, connected : Bool) {
        connectedTo = type
        status = connected
    }
    
    func toMap() -> [String:Any]{
        return [
            "connectedTo" : connectedTo.rawValue,
            "status" : status,
        ]
    }
}

enum NetworkType : String {
    case wifi = "Wifi"
    case cellular = "Cellular"
    case none = "None"
    
    static func fromString(type: String) -> NetworkType{
        switch type {
        case NetworkType.wifi.rawValue:
            return .wifi
        case NetworkType.cellular.rawValue:
            return .cellular
        default: return .none
        }
    }
}

